### PR TITLE
Add switch to filter down reviews to those not approved for first release

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Add filter to filter down to reviews not approved for first release.
Should fix https://github.com/Azure/azure-sdk-tools/issues/4747
NotApprovedForFirstRelease

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Client/src/pages/index.ts [hunk 1]
```diff
@@ -48,6 +49,9 @@ $(() => {
       uri = uri + '&type=' + encodeURIComponent(`${$(this).val()}`);
     });
 
+    if (notApprovedForFirstRelease.prop("checked"))
+      uri = uri + '&notApprovedForFirstRelease=true';
```
Comment (benbp) [inline src/dotnet/APIView/APIViewWeb/Client/src/pages/index.ts:53]: In general I don't recommend single line if statements as it makes it too easy for bugs to creep in.

```suggestion
    if (notApprovedForFirstRelease.prop("checked")) {
      uri = uri + '&notApprovedForFirstRelease=true';
    }
```

File: src/dotnet/APIView/APIViewWeb/Client/src/pages/index.ts [hunk 2]
```diff
@@ -7,7 +7,8 @@ $(() => {
   const languageFilter = $( '#language-filter-select' );
   const stateFilter = $( '#state-filter-select' );
   const statusFilter = $( '#status-filter-select' );
-  const typeFilter = $( '#type-filter-select' );
+  const typeFilter = $('#type-filter-select');
+  const notApprovedForFirstRelease = $('#not-approved-for-first-release');
```
Comment (benbp) [inline src/dotnet/APIView/APIViewWeb/Client/src/pages/index.ts:11]: Should the quote spacing/style be consistent with the above lines?

**Comment by benbp** [comment]
Some questions/suggestions, but LGTM

File: src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs [hunk 3]
```diff
@@ -98,16 +98,52 @@ public async Task<IEnumerable<ReviewModel>> GetRequestedReviews(string userName)
...
+                while (filteredResults.Count < limit && result.Reviews.Count() > 0);
+            }
 
             // Calculate and add Previous and Next and Current page to the returned result
             var totalPages = (int)Math.Ceiling(result.TotalCount / (double)limit);
             var currentPage = offset == 0 ? 1 : offset / limit + 1;
 
             (IEnumerable<ReviewModel> Reviews, int TotalCount, int TotalPages, int CurrentPage, int? PreviousPage, int? NextPage) resultToReturn = (
-                result.Reviews, result.TotalCount, TotalPages: totalPages,
+                filteredResults ?? result.Reviews, result.TotalCount, TotalPages: totalPages,
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs:146]: total pages and totalcount will be incorrect when returning data from filtered results. We need to recalculate them using filtered data.

File: src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs [hunk 4]
```diff
@@ -98,16 +98,52 @@ public async Task<IEnumerable<ReviewModel>> GetRequestedReviews(string userName)
...
+                            }
+                        }
+                    }
+                    if (filteredResults.Count < limit)
+                    {
+                        result = await _reviewsRepository.GetReviewsAsync(
+                            search: search, languages: languages, isClosed: isClosed,
+                            filterTypes: filterTypes, isApproved: isApproved,
+                            notApprovedForFirstRelease: notApprovedForFirstRelease,
+                            offset: offset, limit: limit, orderBy: orderBy);
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs:135]: Shouldn't offset be adjusted when querying for additional reviews to be packed in the filtered result set? Let's discuss about this. I am a bit confused how UI will fetch next set when next page is clicked.

**Pull request closed without merging**